nano/node/socket.cpp

```diff
@@ -62,7 +62,14 @@ void nano::socket::async_connect (nano::tcp_endpoint const & endpoint_a, std::fu
 	this_l->tcp_socket.async_connect (endpoint_a,
 	boost::asio::bind_executor (this_l->strand,
 	[this_l, callback = std::move (callback_a), endpoint_a] (boost::system::error_code const & ec) {
-		this_l->stop_timer ();
+		if (ec)
+		{
+			this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_connect_error, nano::stat::dir::in);
```

Comment:
It is not nano::socket's class fault that the owner of that socket doesn't directly close the socket when receiving a connection error and simply let's it timeout. So in this case, both the connect error and the timeout are correct because that is exactly what is happening, we get a connect error and then the socket is left to timeout. It is not optimal but it is also not a problem.

Reply:
Have a look at `nano::bootstrap_connections::connect_client`, when it receives a connection error, it does nothing and let's the socket timeout. But that is not nano::socket's fault and it is not a problem (just a slight inefficiency) to solve now that I am focusing on nano::socket.